### DISCUSSION TITLE
How to use data from a form to subtract that data from the overall inventory?

### DISCUSSION THREAD

**Discussion by Srij1** [Budibase/budibase#12207] [Help] [answered]
Hi, I'm making an app where it tracks your current inventory, so I have built a screen that lets you make recipes that have certain amounts of ingredients tied to them. I have made a form that asks you what recipe you are using and how much of that recipe you made, so how can I make the data the user provides be subtracted from the overall inventory of the user? What code and where should I write this code? Sorry if this seems very trivial because I am new to Budibase. Thanks!

**Comment by ConorWebb96** [comment, marked as answer]
Hey @Srij1,

This seems like a very good use case for [automations](https://docs.budibase.com/docs/trigger). For example you could use a delete row trigger when a row is delete it takes away from the count within your inventory table. The same could be said for adding to the inventory table. In this case you would use a created/updated trigger and add one to the count whenever a row has been added.

You could even use an app trigger automation, which would pass data into it from the design area. From here you could manipulate that data how you see fit and update the count of the table depending on what you need.

I hope this helps.
